torch/nn/modules/linear.py

```diff
@@ -115,4 +116,99 @@ def __repr__(self):
             + ', in2_features=' + str(self.in2_features) \
             + ', out_features=' + str(self.out_features) + ')'
 
+
+class NoisyLinear(Module):
+    """Applies a noisy linear transformation to the incoming data:
+        :math:`y = (mu_w + sigma_w \cdot epsilon_w)x
+            + mu_b + sigma_b \cdot epsilon_b`
+    More details can be found in the paper `Noisy Networks for Exploration` _ .
+    Args:
+        in_features: size of each input sample
+        out_features: size of each output sample
+        bias: If set to False, the layer will not learn an additive bias.
+            Default: True
+        factorised: whether or not to use factorised noise.
+            Default: True
+        std_init: initialization constant for standard deviation component of
+            weights. If None, defaults to 0.017 for independent and 0.4 for
```

Comment:
Might want to add `(weight_sigma and bias_sigma)` or something similar to make it clearer (that it isn't something like the std dev of noise used to init all weights or something).